src/products/utils/handlers.ts

```diff
@@ -77,23 +77,12 @@ interface ProductAvailabilityArgs {
   productId: string;
 }
 
-export function getProductAvailabilityVariables({
+export const getProductAvailabilityVariables = ({
   isAvailableForPurchase,
   availableForPurchase,
   productId
-}: ProductAvailabilityArgs) {
-  const isAvailable =
-    availableForPurchase && !isAvailableForPurchase
-      ? true
-      : isAvailableForPurchase;
-
-  return {
-    isAvailable,
-    productId,
-    startDate: isAvailableForPurchase
-      ? null
-      : availableForPurchase !== ""
-      ? availableForPurchase
-      : null
-  };
-}
+}: ProductAvailabilityArgs) => ({
+  isAvailable: isAvailableForPurchase,
+  productId,
+  startDate: availableForPurchase
```

Comment:
`startDate` should be set to null if `isAvailableForPurchase` is set to true,
(or maybe it should be changed in ProductUpdatePage)

Reply:
Actually not. The idea behind this PR is because this piece of logic was changed for 2.11 - when sending a date for available / published the isAvailable is also to be set to true